### PR TITLE
improved string normalization for deduplication

### PR DESCRIPTION
**Pull request by missinglink**
this PR improves the normalization function used for deduplication:
- remove diacritics
- improve test coverage

### CODE REVIEW

**Comment by missinglink** [comment]
I staged this to take a look and there were three notable changes:

```
Aggregate test results
Pass: 610
Improvements: 1
Fail: 115
Placeholders: 0
Regressions: 2
Took 114607ms
Test success rate 99.72%
```

I had a look and it seems as though they are actually positive but marked as failing due to brittle test cases.

```
General Placeholder requests
  ✘ regression [17] "{"text":"Schöneberg, DE"}": score 2 out of 3
  diff:
    gid
      expected: whosonfirst:locality:101907543
      actual:   whosonfirst:locality:1126037037

search city/country
 ✔ improvement [15] "{"text":"Akmenė, lithuania"}"

Who's on First regions
  ✘ regression [8] "{"text":"Xaçmaz, Azerbaijan","sources":"wof"}": score 3 out of 5
  diff:
    layer
      expected: region
      actual:   locality
    name
      expected: Xacmaz
      actual:   Xaçmaz
```

**Comment by orangejulius** [comment]
Looks good to me, is there anything else to do here before merging?

File: test/unit/helper/diffPlaces.js [hunk 1]
```diff
@@ -337,6 +339,104 @@ module.exports.tests.dedupe = function(test, common) {
...
+    t.true(normalizeString('foo, bar'), 'foo bar');
+    t.true(normalizeString('foo-bar'), 'foo bar');
+    t.true(normalizeString('foo , - , - bar'), 'foo bar');
+    t.end();
+  });
+
+  test('diacritics', function (t) {
+    t.true(normalizeString('Malmö'), 'malmo');
+    t.true(normalizeString('Grolmanstraße'), 'grolmanstraße');
+    t.true(normalizeString('àáâãäåấắæầằçḉèéêëếḗềḕ'), 'aaaaaaaaaeaacceeeeeeee');
```
Comment (missinglink) [inline test/unit/helper/diffPlaces.js]: These tests should be `t.equals` not `t.true`

File: test/unit/helper/diffPlaces.js [hunk 2]
```diff
@@ -337,6 +339,104 @@ module.exports.tests.dedupe = function(test, common) {
...
+    ), 'Malmö');
+
+    t.end();
+  });
+};
+
+module.exports.tests.normalizeString = function (test, common) {
+  test('lowercase', function (t) {
+    t.true(normalizeString('Foo Bar'), 'foo bar');
+    t.true(normalizeString('FOOBAR'), 'FOOBAR');
```
Comment (missinglink) [inline test/unit/helper/diffPlaces.js]: The right side of this test should appear in lowercase

**Comment by missinglink** [comment]
I fixed some test issues and also now use our fancy new `unicode.normalize()` method to fit any potential unicode issues 🎉 

This is good to merge once the failing test cases I mentioned above have been investigated.

**Comment by missinglink** [comment]
force pushed to rebase `origin/master`

**Comment by missinglink** [comment]
**before**

```
Aggregate test results
Pass: 615
Improvements: 10
Fail: 123
```

**after**

```
Aggregate test results
Pass: 645
Improvements: 5
Fail: 128
```

---

There is one notable test failure but I switched to `dev->api` to the same hash as `prod` and it's there too, so it's not the fault of this work 👍 

```
search focus.point
  ✘ regression [1] "/v1/search?focus.point.lat=42.64&focus.point.lon=-84.7&text=10010": score 3 out of 4
  diff:
    priorityThresh is 1 but found at position 2
```

https://pelias.github.io/compare/#/v1/search?text=10010

**Comment by missinglink** [comment]
The three failing test cases mentioned in the previous comment are no longer failing.

**Pull request merged**